src/main/scala/performanceanalysis/ParsedLine.scala

```diff
@@ -0,0 +1,66 @@
+package performanceanalysis
+
+import java.time.LocalDateTime
+import java.time.format.{DateTimeFormatter, DateTimeFormatterBuilder, ResolverStyle}
+import java.time.temporal.ChronoField
+
+import scala.util.matching.Regex
+import scala.util.matching.Regex.Match
+
+object LineParser {
+
+  def apply(regex: Regex): LineParser = new LineParser(regex)
+
+  def apply(string: String): LineParser = apply(string.r)
+
+}
+
+class LineParser(regex: Regex) {
+
+  def parse(line: String): ParsedLine = new ParsedLine(line, regex)
+
+}
+
+object ParsedLine {
+
+  val dtRegex = """(\d{1,2})[ -/.](\d{1,2})[ -/.](\d{4})[ T]( \d|\d{2})[:.](\d{2})[.:](\d{2})(\.\d{0,9})?""".r
+
+  def dateParser(iYear: Int, iMonth: Int, iDay: Int): String => Option[LocalDateTime] = {
+    (s: String) => try {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using util.Try instead of a try-catch block](https://www.codacy.com/app/scala-academy/performance-analysis/file/2589632303/issues/source?bid=3220915&fileBranchId=3304352#l29)